### DISCUSSION TITLE
How can I get information about fetching next page error?

### DISCUSSION THREAD

**Discussion by VKhayretdinov** [TanStack/query#2244] [General]
Hello!

React Query allows you to get information only about a loading error or a fetching error, but doesn't provide separate information about getting an error while fetching the next page.

My current solution is to use information about fetch direction and refetchError:
  ```
const queryResult = useInfiniteQuery(queryKey, queryFn);

  const queryClient = useQueryClient();
  const queryFoundResult = queryClient.getQueryCache().find(queryKey);
  
  queryFoundResult?.state?.fetchMeta?.fetchMore?.direction === 'forward' && queryResult.isRefetchError
```

Is there a better solution for this case?

Thanks!

**Comment by TkDodo** [comment]
Not sure I understand what a loading error or a fetching error is. As far as I know, if you fetch the next page and it errors, the whole query will go to error state, so `isError` will be true and the `error` prop will be set

**Reply by VKhayretdinov** [reply]
I mean `useInfiniteQuery(queryKey, queryFn)` returns `isLoadingError` and `isRefetchError`.

> if you fetch the next page and it errors, the whole query will go to error state, so isError will be true and the error prop will be set

You're right, but this doesn't allow me to understand the type of the error: Did I get an error during a simple background fetch (like every 1 minute), or did I get an error when loading the next page
